src/App.tsx

```diff
@@ -149,6 +149,8 @@ class App extends React.Component<any, IState> {
         <Toolbar
           url={this.state.url}
           onActionInvoked={this.onToolbarActionInvoked}
+          onClipboardRead={this.onClipboardRead.bind(this)}
```

Comment:
Why is the clipboard actions treated differently than the regular actions?

Reply:
The reason was that reading from clipboard needs to return a value, so not like the current `onToolbarActionInvoked` signature, the reading method needs to have a `Promise<string>` object as its return value (and I preferred to implement it with a strictly typed return value instead of `Promise<any>`). But, you are totally right it would be more concise if we just use onToolbarActionInvoked. I'm going to push the appropriate commit right now.